### ISSUE TITLE
Binary messaging: values > a nibble become 253

### ISSUE THREAD

**Issue by rgbkrk** [ipython/ipywidgets#228]
As stated in https://github.com/rgbkrk/bitjet/pull/22, using `Bytes` from traitlets with ipywidgets turns byte values >= 128 into 253 across the board.

**Comment by jdfreder** [comment]
Whoops, I failed to use the search function.  Closing #229 in favor of this.

**Comment by rgbkrk** [comment]
GAH! You got 229, my favorite number!!!

**Comment by minrk** [comment]
We aren't using the binary serialization for Bytes traits, which is probably the cause of the problem. The Bytes are being cast to str, which is probably getting normalized due to failed text encoding.

I opened rgbkrk/bitjet#23, which tells it to use actual binary messages via wrapping the bytes in a memoryview. While investigation, I also opened #231, which fixes bugs that made binary messages impossible to send with current master, and triggers binary messages for Bytes by default.
